src/Interpreters/executeQuery.cpp

```diff
@@ -897,6 +897,14 @@ void validateAnalyzerSettings(ASTPtr ast, bool context_value)
 
     bool top_level = context_value;
 
+    auto field_to_bool = [&](Field & f) -> bool
+    {
+        if (f.getType() == Field::Types::String)
+            return stringToBool(f.safeGet<String>());
+        else
+            return f.safeGet<bool>();
+    };
```

Comment:
```suggestion
    auto field_to_bool = [](const Field & f) -> bool
    {
        if (f.getType() == Field::Types::String)
            return stringToBool(f.safeGet<String>());
        else
            return f.safeGet<bool>();
    };
```

I think we can remove `&` capture and pass `f` as const.

Reply:
Good catch, applied, thanks!